src/components/form/fields/CodeField.tsx

```diff
@@ -1,11 +1,54 @@
 import i18n from '@dhis2/d2-i18n'
-import { InputFieldFF } from '@dhis2/ui'
-import React from 'react'
+import { InputFieldFF, Validator } from '@dhis2/ui'
+import React, { useMemo } from 'react'
 import { Field as FieldRFF } from 'react-final-form'
-import { SchemaSection, useCheckMaxLengthFromSchema } from '../../../lib'
+import { useParams } from 'react-router-dom'
+import {
+    composeAsyncValidators,
+    required,
+    SchemaFieldPropertyType,
+    SchemaSection,
+    useIsFieldValueUnique,
+    useSchema,
+} from '../../../lib'
+import { useCheckMaxLengthFromPropriety } from '../../../lib/models/useCheckMaxLengthFromSchema'
+function useValidator({
```

Comment:
Maybe we should extract this to it's own file, and re-use it since its now more generic?

Reply:
kk doen it for codeField. Let me know what you think and we can substitute the others as we go